src/stdlib/math.rs

```diff
@@ -18,9 +20,30 @@ pub fn new() -> Module {
             "nan".into() => Type::Float(f32::NAN),
 
             // funcs
-            "sin".into() => native_func!(|_, _args| {
-                Ok(Type::Float(3f32))
-            }, 1)
+            // TODO: add more functions
+            // Trigonometric functions
+            "sin".into() => native_func!(|_, args| {
+                Ok(Type::Float(args[0].parse::<f32>()
+                    .expect("Invalid Argument: Type is not Float").sin()))
+            }, 1),
+            "cos".into() => native_func!(|_, args| {
+                Ok(Type::Float(args[0].parse::<f32>()
+                    .expect("Invalid Argument: Type is not Float").cos()))
+            }, 1),
+            "tan".into() => native_func!(|_, args| {
+                Ok(Type::Float(args[0].parse::<f32>()
+                    .expect("Invalid Argument: Type is not Float").tan()))
+            }, 1),
+
+            // RNG functions
+            "randrange".into() => native_func!(|_, args| {
+                Ok(Type::Float(thread_rng().gen_range(
+                    args[0].parse::<i32>()
+                        .expect("Invalid Argument: Type is not Int")..
+                    args[1].parse::<i32>()
+                        .expect("Invalid Argument: Type is not Int")
+                ) as f32))
+            }, 2)
```

Comment:
I don't like how these will panic when the argument is wrong. These kinds of errors should be handled within the language, and catchable. Could you instead try to ensure the argument is of `Type::Float`?

Reply:
Ah, I see. I just haven't analyzed the whole language interpreter, so I haven't realized that there is a way to raise error without just making the program panic. Definitely will change those to TypeError-throws

Reply:
If that is the case, I'd ask that you do check carefully as to how the overall system works, at least for the general types and errors. (CC @cursorweb: we should really have some developer documentation)

Reply:
ok

Reply:
Solved! Kinda.

Reply:
This was not resolved.